packages/core/src/modules/cell.ts

```diff
@@ -284,6 +284,21 @@ export function setCellValue(
     } else if (!_.isNil(cell.ct) && cell.ct.fa === "@") {
       cell.m = vupdateStr;
       cell.v = vupdate;
+    } else if (!_.isNil(cell.ct) && !_.isNil(cell.ct.t) && cell.ct.t === "d") {
+      const mask = genarate(vupdate);
+      if (_.isNil(mask)) {
+        cell.m = !_.isNil(cell.ct.fa)
+          ? update(cell.ct.fa, vupdate).toString()
+          : vupdate;
+        cell.v = vupdate;
+        return;
+      }
+      if (mask?.[1].fa === cell.ct.fa) {
+        [cell.m, , cell.v] = mask;
+        return;
+      }
+      cell.m = vupdate;
+      cell.v = vupdate;
```

Comment:
```suggestion
    } else if (cell.ct != null && cell.ct.t === "d" && _.isString(vupdate)) {
      const mask = genarate(vupdate) as any;
      if (mask[1].t !== "d" || mask[1].fa === cell.ct.fa) {
        [cell.m, cell.ct, cell.v] = mask;
      } else {
        cell.v = mask[2];
        cell.m = update(cell.ct.fa!, cell.v);
      }
```

* `mask` is only null when the input is null, so just check if the input is string.
* If the input is not a valid date, update the `ct` and values
* If the input is a valid date, and the resulting `fa` is the same, just copy the values
* If the input is a valid date, but `fa` is different, re-format it